File: 2022/Day1/sln.hs
import Data.List.Split
import Data.List

main = do 
    calories <- readFile "Day1/input.txt"
    print $ map (takeMax calories) [1,3]

parseToInt = map (\y -> read y :: Integer) 

takeMax x n = sum $ take n $ reverse $ sort $ map (sum . parseToInt) $ splitWhen (=="") $ lines x

File: 2022/Day1/sln.rs
use std::fs;

fn main() {
    let contents = fs::read_to_string("./Day1/input.txt").unwrap();
    let calories = contents.split("\n\n")
        .map(|x| x.lines().map(|x|
             x.parse::<i32>().unwrap()));
    let mut total_cals: Vec<i32> = calories.map(|x| x.sum::<i32>()).collect();
    total_cals.sort_by(|a,b| b.cmp(a));
    print!("Part 1: {}\n", total_cals[0]);
    print!("Part 2: {}\n", total_cals[..=2].iter().sum::<i32>())
}